pydatastructs/linear_data_structures/linked_lists.py

```diff
@@ -581,3 +583,133 @@ def extract(self, index):
         elif index == 0:
             self.tail.next = self.head
         return node
+
+class SkipList(object):
+    """
+    Represents Skip List
+
+    Examples
+    ========
+
+    >>> from pydatastructs import SkipList
+    >>> sl = SkipList()
+    >>> sl.insert(6)
+    >>> sl.insert(1)
+    >>> sl.insert(3)
+    >>> repr(sl)
+    '-inf.->1.->3.->6.->inf.'
+    >>> sl.remove(1)
+    >>> sl.insert(4)
+    >>> sl.insert(2)
+    >>> sl.search(4)
+    True
+    >>> sl.search(10)
+    False
+    >>> repr(sl)
+    '-inf.->2.->3.->4.->6.->inf.'
+
+    """
+
+    def __new__(cls):
+        obj = object.__new__(cls)
+        obj.head = None
+        obj.tail = None
+        obj._addOneLevel()
+        return obj
+
+    def __repr__(self):
+        li = self.__str__()
+        return '->'.join(li[-1])
+
+    def __vars__(self):
+        li = self.__str__()
+        return ' '.join(map(lambda l: '->'.join(l), li))
```

Comment:
Why these extra functions?

Reply:
vars is dynamic if the user want to print the whole skiplist with each layer it can do so.